examples/bri-3/src/bri/zeroKnowledgeProof/api/ccsmAnchorHash.controller.spec.ts

```diff
@@ -0,0 +1,132 @@
+import { BadRequestException } from '@nestjs/common';
+import { CqrsModule } from '@nestjs/cqrs';
+import { Test, TestingModule } from '@nestjs/testing';
+import { INVALID_ANCHOR_HASH_INPUT } from './err.messages';
+import { CCSMAnchorHashController } from './ccsmAnchorHash.controller';
+import { CCSMAnchorHashAgent } from '../agents/ccsmAnchorHash.agent';
+import { CreateCCSMAnchorHashCommandHandler } from '../capabilities/createCCSMAnchorHash/createCCSMAnchorHashCommand.handler';
+import { VerifyCCSMAnchorHashCommandHandler } from '../capabilities/verifyCCSMAnchorHash/verifyCCSMAnchorHashCommand.handler';
+import { CCSMAnchorHashStorageAgent } from '../agents/ccsmAnchorHashStorage.agent';
+import { CreateCCSMAnchorHashDto } from './dtos/request/createCCSMAnchorHash.dto';
+import { VerifyCCSMAnchorHashDto } from './dtos/request/verifyCCSMAnchorHash.dto';
+import { BpiSubjectType } from '../../identity/bpiSubjects/models/bpiSubjectType.enum';
+import { BpiSubject } from '../../identity/bpiSubjects/models/bpiSubject';
+import { BpiSubjectAccount } from '../../identity/bpiSubjectAccounts/models/bpiSubjectAccount';
+import { BlockchainService } from '../services/blockchain/blockchain.service';
+
+describe('ProofController', () => {
+  let controller: CCSMAnchorHashController;
+
+  beforeEach(async () => {
+    const app: TestingModule = await Test.createTestingModule({
+      imports: [CqrsModule],
+      controllers: [CCSMAnchorHashController],
+      providers: [
+        CCSMAnchorHashAgent,
+        CreateCCSMAnchorHashCommandHandler,
+        VerifyCCSMAnchorHashCommandHandler,
+        CCSMAnchorHashStorageAgent,
+        BlockchainService,
+      ],
+    }).compile();
+
+    controller = app.get<CCSMAnchorHashController>(CCSMAnchorHashController);
+
+    await app.init();
+  });
+
+  describe('createCCSMAnchorHash', () => {
+    it('should throw BadRequest if document parameter is empty', async () => {
+      // Arrange
+      const mockBpiSubject = new BpiSubject(
+        '123',
+        '123',
+        '123',
+        BpiSubjectType.External,
+        '123',
+      );
+
+      const mockBpiSubjectAccount = new BpiSubjectAccount(
+        '123',
+        mockBpiSubject,
+        mockBpiSubject,
+      );
+
+      const mockDocument = '';
+
+      const missingDocumentParam = {
+        ownerAccount: mockBpiSubjectAccount,
+        document: mockDocument,
+      };
+
+      // Act and assert
+      await expect(async () => {
+        await controller.createCCSMAnchorHash(missingDocumentParam);
```

Comment:
i think this dto (CreateCCSMAnchorHashDto) can be simplified to only contain ownerId instead of whole owner